stats/memstats/stats.go

```diff
@@ -211,6 +226,56 @@ func (ms *Store) Get(name string, tags stats.Tags) *Measurement {
 	return ms.byKey[ms.getKey(name, tags)]
 }
 
+// GetAll returns the metric for all name/tags register in the store.
+func (ms *Store) GetAll() []Metric {
+	return ms.getAllByName("")
+}
+
+// GetByName returns the metric for each tag variation with the given name.
+func (ms *Store) GetByName(name string) []Metric {
```

Comment:
Does it make sense to have `GetByName` return a slice instead of just `Metric` given the current implementation? Perhaps if we were to use `Index` instead of a hard comparison it might make more sense:

Instead of this:
```go
keys := lo.Filter(lo.Keys(ms.byKey), func(k string, index int) bool {
	return name == "" || ms.byKey[k].name == name
})
```

Do this:
```go
keys := lo.Filter(lo.Keys(ms.byKey), func(k string, index int) bool {
	return name == "" || strings.Index(ms.byKey[k].name, name) != -1
})
```

Wdyt?

Reply:
Sorry @fracasula, I didn't get your point here.

`GetByName` can return multiple metrics, even with the same name you can have multiple tag combinations.